modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/ValueProviderUtils.java

```diff
@@ -39,6 +40,18 @@ private ValueProviderUtils() {
 
   }
 
+  static org.mule.runtime.extension.api.values.ValueBuilder cloneAndEnrichValue(org.mule.runtime.api.value.Value value,
```

Comment:
why is this needed? Doesn't the need for this disappears if adaptation is done in a single place and in the correct direction?

Reply:
Tooling client uses this utils

Reply:
They shouldn't. Please create a ticket for us to provide an API for whatever internal functionality they are using and add a TODO here.